doc/porting/shields.rst

```diff
@@ -0,0 +1,83 @@
+.. _shields
+
+Shields
+#######
+
+Shields have been introduced on Zephyr as a new element that can be used
+for easier prototyping.
+
+Aim of the shield feature is to provide zephyr formatted shield description
+for easier compatibility with applications.
+
+Shield porting and configuration
+********************************
+
+Shields are available in board directory under :file:`/boards/shields`.
+Following files are provided for shield configuration:
+
+.. code-block:: none
+
+   boards/shields/<shield>
+   ├── Kconfig.shield
+   ├── Kconfig.defconfig
+   ├── <shield>.overlay
+   └── dts.fixup
+
+These files provides shield configuration as follows:
+
+* **Kconfig.shield**: This file defines the shield's config flag
+  (CONFIG_SHIELD_X). This flag will be used in an application to trigger the
+  shield configuration on top of board configuration. Besides in this file,
+  each driver of shield components (sensors for instance) are enabled.
+
+* **Kconfig.defconfig**: This file is used to provide specific driver default
+  configuration. Since congiruration could be done conditionally it provides
+  flexibility for broader board support.
+
+* **<shield>.overlay**: Device tree overlay is a device tree fragment that
+  provides shield description in device tree format. This file will be merged
+  with board device tree file before compilation.
+
+* **dts.fixup**: fixup file to bind board components definitions with
+  application in a generic fashion to enable shield compatibility across boards
+
+Board compatibility
+*******************
+
+Shields to board compatibility relies on use of well-known connectors populated
+in various boards (Arduino, 96boards, etc). In order to enable this
+compatibility at software level, boards need to provide a configuration matching
+the connector they support.
+
+This should be done at two different level:
+
+* Pinmux: Connector pins should be correctly configured to match shield pining
+
+* Device tree: Board device tree file should propose a node alias for each
```

Comment:
A board device tree file should define a node alias for each
    connector interface.  For example, for Arduino I2C: